### PR TITLE
Add ClientCLI

### PR DESCRIPTION
**Pull request by sb10**
Make it easier for clients to create a CLI where they can store user jwt and server token on disk and use those for login automatically.

Breaking change to some public APIs.

### CODE REVIEW

File: cli.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
+/*******************************************************************************
+ * Copyright (c) 2022-2024 Genome Research Ltd.
+ *
+ * Authors:
+ *	- Sendu Bala <sb10@sanger.ac.uk>
+ *	- Michael Grace <mg38@sanger.ac.uk>
```
Comment (mjkw31) [inline cli.go:6]: This is a new file, right? Don't think mg38 worked on this.

**Review by mjkw31 — CHANGES_REQUESTED** [review]

File: cli.go (hunk 1 above)
Comment (sb10) [reply cli.go:6]: > This is a new file, right? Don't think mg38 worked on this.

It's largely copied from code mg38 was involved with in another repo, so I've left that in.

Comment (mjkw31) [reply cli.go:6]: Fair enough.

**Review by mjkw31 — APPROVED** [review]
LGTM.

**Pull request merged**